packages/transaction-controller/src/utils.ts

```diff
@@ -231,3 +244,51 @@ export function transactionMatchesNetwork(
   }
   return false;
 }
+
+/**
+ * Validates the external provided transaction meta.
+ *
+ * @param transactionMeta - The transaction meta to validate.
+ * @param confirmedTxs - The confirmed transactions in controller state.
+ * @param pendingTxs - The submitted transactions in controller state.
+ */
+export function validateConfirmedExternalTransaction(
```

Comment:
I'm torn on the benefit of this `utils` file as the majority of functions are not exported by the package but only used internally by the `TransactionController`.

This means that it's hard to differentiate them from the private methods in the controller itself as many such as this one are only used in a single flow, so we're not even simplifying the unit testing but instead segmenting it and testing internals directly.

Would it be cleaner to move all the new private and utility methods you've added into an `external-transactions.ts` file with a comment to say all of the functions are used exclusively by `confirmExternalTransaction` and are tested in context.

We could also then move the associated tests for these functions into `TransactionController.test.ts` with the context of calling `confirmExternalTransaction`?

Reply:
I see your point, that make sense to me. I will take this into account on my upcoming PRs

Addressed in: https://github.com/MetaMask/core/pull/1625/commits/33c08f4d3cddf3abee8826ae8df06bd84026377b